src/Employer/Employer.Web/Filters/LevyDeclarationCheckFilter.cs

```diff
@@ -0,0 +1,112 @@
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Logging;
+
+namespace Esfa.Recruit.Employer.Web.Filters
+{
+    public class LevyDeclarationCheckFilter : IAsyncActionFilter, IOrderedFilter
+    {
+         private readonly IEmployerVacancyClient _vacancyClient;
+        private readonly IDataProtector _dataProtector;
+        private readonly IHostingEnvironment _hostingEnvironment;
+        private readonly ILogger<LevyDeclarationCheckFilter> _logger;
+
+        public LevyDeclarationCheckFilter(IEmployerVacancyClient vacancyClient,
+            IDataProtectionProvider dataProtectionProvider,
+            IHostingEnvironment hostingEnvironment,
+            ILogger<LevyDeclarationCheckFilter> logger)
+        {
+            _vacancyClient = vacancyClient;
+            _dataProtector = dataProtectionProvider.CreateProtector(DataProtectionPurposes.LevyDeclarationCookie);
+            _hostingEnvironment = hostingEnvironment;
+            _logger = logger;
+        }
+
+        public int Order { get; } = 50;
+
+        public async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+        {
+            if (HasValidLevyCookie(context))
+            {
+                if (RequestIsForALevyPage(context))
+                {
+                    // Redirect to dashboard as they've already chosen a levy option
+                    var employerAccountId = context.RouteData.Values[RouteValues.EmployerAccountId].ToString().ToUpper();
+                    context.Result = new RedirectToRouteResult(RouteNames.Dashboard_Index_Get, new { employerAccountId });
+                    
+                    return;
+                }
+            }
+            else if (!RequestIsForWhiteListedPage(context))
+            {
+                var userId = context.HttpContext.User.GetUserId();
+                var details = await _vacancyClient.GetUsersDetailsAsync(userId);
+
+                if (details.DeclaredAsLevyPayer)
+                {
+                    var protectedUserId = _dataProtector.Protect(userId);
+                    context.HttpContext.Response.Cookies.Append(CookieNames.LevyEmployerIndicator, protectedUserId, EsfaCookieOptions.GetDefaultHttpCookieOption(_hostingEnvironment));
+                }
+                else
+                {
+                    var employerAccountId = context.RouteData.Values[RouteValues.EmployerAccountId].ToString().ToUpper();
+                    
+                    // Redirect to Levy page
+                    context.Result = new RedirectToRouteResult(RouteNames.LevyDeclaration_Get, new { employerAccountId });
+                    
+                    return;
+                }
+            }
+
+            await next(); 
+        }
+
+        private bool RequestIsForWhiteListedPage(ActionExecutingContext context)
```

Comment:
Would an Ignore attribute work here?
https://stackoverflow.com/questions/9953760/how-to-disable-a-global-filter-in-asp-net-mvc-selectively

Reply:
We need to do more than skip in the circumstance where they have the cookie and are trying to navigate to the levy declaration page. We want to redirect to dashboard as opposed to skipping.